custom_components/homewhiz/api.py

```diff
@@ -0,0 +1,150 @@
+import datetime
+import hashlib
+import hmac
+import json
+import logging
+
+import aiohttp
+from aiohttp import ContentTypeError
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+class RequestError(Exception):
+    pass
+
+
+class LoginError(Exception):
+    pass
+
+
+def sign(key, msg):
+    return hmac.new(key, msg.encode("utf-8"), hashlib.sha256).digest()
+
+
+def getSignatureKey(key, date_stamp, regionName, serviceName):
+    kDate = sign(("AWS4" + key).encode("utf-8"), date_stamp)
+    kRegion = sign(kDate, regionName)
+    kService = sign(kRegion, serviceName)
+    kSigning = sign(kService, "aws4_request")
+    return kSigning
+
+
+async def login(username: str, password: str):
+    request_parameters = {"password": password, "username": username}
+
+    async with aiohttp.ClientSession() as session:
+        async with session.post(
+            "https://api.arcelikiot.com/auth/login",
+            json=request_parameters,
+        ) as response:
+            contents = await response.json()
+            if not contents["success"]:
+                _LOGGER.error(json.dumps(contents, indent=4))
+                raise LoginError(contents)
+            data = contents["data"]
+            return data["credentials"]
+
+
+async def make_id_exchange_request(device_name: str):
+    hsmid = device_name[4:]
+    _LOGGER.debug(f"hsmid: {hsmid}")
+    async with aiohttp.ClientSession() as session:
+        async with session.get(
+            f"https://idexchange.arcelikiot.com/GetApplianceId?hsmid={hsmid}",
+        ) as response:
+            if not response.ok:
+                _LOGGER.error(await response.text())
+                raise RequestError()
+            contents = json.loads(await response.text())
+            return contents
```

Comment:
I personally would return appId and modelId here directly + some more documentation (type annotations, docstrings) would be nice (in general).
 ```python
appId = id_response["appId"]
modelId = id_response["m"]
```